geowrangler/datasets/nightlights.py

```diff
@@ -0,0 +1,526 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: notebooks/14_datasets_nightlights.ipynb (unless otherwise specified).
+
+__all__ = [
+    "get_eog_access_token",
+    "clear_eog_access_token",
+    "setup_eog_auth_headers",
+    "download_url",
+    "unzip_eog_gzip",
+    "clip_raster",
+    "URLFORM",
+    "EOG_VIIRS_DATA_TYPE",
+    "EOG_PRODUCT",
+    "EOG_PRODUCT_VERSION",
+    "EOG_COVERAGE",
+    "make_url",
+    "make_clip_hash",
+    "generate_clipped_raster",
+    "generate_clipped_metadata",
+    "get_clipped_raster",
+    "generate_nightlights_feature",
+]
+
+
+# Internal Cell
+
+
+import gzip
+import hashlib
+import json
+import os
+import shutil
+from pathlib import Path
+from types import SimpleNamespace
+from urllib.error import HTTPError
+from urllib.parse import urlparse
+
+import numpy as np
+import requests
+from fastcore.net import urldest
+from loguru import logger
+from shapely.geometry import box
+
+import geowrangler.raster_process as rp
+import geowrangler.raster_zonal_stats as rzs
+from .utils import make_report_hook, urlretrieve
+
+# Internal Cell
+DEFAULT_EOG_CREDS_PATH = "~/.eog_creds/eog_access_token"
+EOG_ENV_VAR = "EOG_ACCESS_TOKEN"
+NIGHTLIGHTS_CACHE_DIR = "~/.cache/geowrangler/nightlights"
+
+# Cell
+
+
+# Retrieve access token
+def get_eog_access_token(
+    username,
+    password,
+    save_token=False,
+    save_path=DEFAULT_EOG_CREDS_PATH,
+    set_env=True,
+    env_token_var=EOG_ENV_VAR,
+):
+    params = {
+        "client_id": "eogdata_oidc",
+        "client_secret": "2677ad81-521b-4869-8480-6d05b9e57d48",
```

Comment:
I assume this is safe and wont have us flagged by eog for downloading their data via this?

Reply:
Yes. See their [suggested python example](https://eogdata.mines.edu/products/register/#programmed_access)